broker/lib/fabrik/DirectorManager.js

```diff
@@ -260,9 +260,12 @@ class DirectorManager extends BaseManager {
       .set('plan_id', this.plan.id)
       .set('agent_properties', _.omit(agentProperties, 'auth', 'provider'))
       .value();
+    const agentCredsBeforeUpdate = utils.getBrokerAgentCredsFromManifest(context.params.previous_manifest);
```

Comment:
Its better if this is moved to agent. Preupdate method anyways is sent with context. Director need not be aware of this. This logic should be within agent. i.e. which creds to use.

Reply:
@jagadish-kb  : Didn't get you . are you saying that the /info call returns the creds to be used by broker while calling the /preupdate API ?

Reply:
what i meant is the line 263, can be pushed into the function Agent.preUpdate(). Reason being, directorManager should not care what creds are to be used and ideally thats something that must be encapsulated within the Agent class. Dont want that responsibility spilling outside of it.

Reply:
Ok , got it . Yes , it makes sense. When you said agent , thought it was service agent :)

Reply:
Makes sense. I'll make the change.

Reply:
Done!